### PR TITLE
network healthcheck changes

### PR DESCRIPTION
**Pull request by leodotcloud**
https://github.com/rancher/rancher/issues/10326

### CODE REVIEW

File: code/packaging/app-config/src/main/resources/META-INF/cattle/system-services/defaults.properties [hunk 1]
```diff
@@ -69,6 +69,8 @@ metadata.increment.processes=account.update, \
     hostipaddressmap.activate, \
     host.remove, \
     host.update, \
+    host.activate, \
+    host.deactivate, \
     instancehostmap.activate, \
```
Comment (cjellick) [inline code/packaging/app-config/src/main/resources/META-INF/cattle/system-services/defaults.properties:82]: @alena1108 any bad repercussions from adding host activate and decactivate to metadata.increment.process?

File: code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/MetaDataInfoDaoImpl.java [hunk 2]
```diff
@@ -219,7 +219,7 @@ protected HostMetaData map(List<Object> input) {
             }
         };
 
-        HostTable host = mapper.add(HOST);
+        HostTable host = mapper.add(HOST, HOST.STATE);
```
Comment (cjellick) [inline code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/MetaDataInfoDaoImpl.java]: My understanding is that add(HOST) would add all the fields from host, including state. Is that not the case?

Comment (leodotcloud) [reply code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/MetaDataInfoDaoImpl.java]: I guess you are right, will remove this line and test.

File: code/packaging/app-config/src/main/resources/META-INF/cattle/system-services/defaults.properties (hunk 1 above)
Comment (alena1108) [reply code/packaging/app-config/src/main/resources/META-INF/cattle/system-services/defaults.properties:82]: @cjellick not really. We already have plenty of processes triggering meta update...hard to make it worse. Plus activate/deactivate doesn't happen too often.

File: code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/serviceevent/ServiceEventFilter.java [hunk 3]
```diff
@@ -114,10 +121,23 @@ private boolean isNetworkUp(long accountId) {
...
+            if (!(service.getState().equals(CommonStatesConstants.ACTIVE) && service.getHealthState().equals(HealthcheckConstants.HEALTH_STATE_HEALTHY))) {
                 return false;
             }
         }
         return true;
     }
+
+    private boolean isNetworkStack(long accountId, long instanceId) {
+        Instance instance = objectManager.findAny(Instance.class, INSTANCE.ID, instanceId);
+        List<? extends Service> services = instanceDao.findServicesFor(instance);
```
Comment (alena1108) [inline code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/serviceevent/ServiceEventFilter.java]: see if we can change this method in instanceDao to accept instanceid (I bet instanceid is what get used in db query anyway). This way we don't have to load instance on the line 132.

File: code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/serviceevent/ServiceEventFilter.java [hunk 4]
```diff
@@ -114,10 +121,23 @@ private boolean isNetworkUp(long accountId) {
...
+        if(services.size() > 0) {
+            List<Service> network_services = objectManager.find(Service.class, SERVICE.ACCOUNT_ID, accountId, SERVICE.REMOVED, null, SERVICE.STACK_ID,
```
Comment (alena1108) [inline code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/serviceevent/ServiceEventFilter.java]: You already have the list of services for the instance on line 133, no need to look for them again. What you need to do is - locate the services having kind == network for the instance. You can introduce a new method for that in instanceDao that would accept instanceid and servicekind

File: code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java [hunk 5]
```diff
@@ -176,6 +176,15 @@ public Instance getInstanceByUuidOrExternalId(Long accountId, String uuid, Strin
                 .where(SERVICE_EXPOSE_MAP.INSTANCE_ID.eq(instance.getId()))
                 .fetchInto(ServiceRecord.class);
     }
+    
+    public List<? extends Service> findServicesForInstanceId(Long instanceId){
+    		return create().select(SERVICE.fields())
+    				.from(SERVICE)
+    				.join(SERVICE_EXPOSE_MAP)
+    				.on(SERVICE_EXPOSE_MAP.SERVICE_ID.eq(SERVICE.ID))
```
Comment (alena1108) [inline code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java]: should check for removed being null on both service and serviceexposemap

File: code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java [hunk 6]
```diff
@@ -176,6 +176,15 @@ public Instance getInstanceByUuidOrExternalId(Long accountId, String uuid, Strin
                 .where(SERVICE_EXPOSE_MAP.INSTANCE_ID.eq(instance.getId()))
                 .fetchInto(ServiceRecord.class);
     }
+    
+    public List<? extends Service> findServicesForInstanceId(Long instanceId){
```
Comment (alena1108) [inline code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java]: should be long primitive

**Comment by alena1108** [comment]
@kinarashah can we squash 2 commits of yours into 1?

File: code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java (hunk 5 above)
Comment (kinarashah) [reply code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java]: https://github.com/leodotcloud/cattle/blob/e27b2641a783d877d6480644606ff442cd5cc5d8/code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/InstanceDaoImpl.java#L171

should it be added for `findServicesFor` as well then?

**Comment by alena1108** [comment]
LGTM

**Pull request merged**